src/main/java/mate/dao/impl/DriverDaoImpl.java

```diff
@@ -109,4 +112,32 @@ private Driver getDriver(ResultSet resultSet) throws SQLException {
         driver.setId(id);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT * FROM drivers WHERE login = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            Driver driver = null;
+            if (resultSet.next()) {
+                driver = getDriverWithLoginAndPassword(resultSet);
+            }
+            return Optional.ofNullable(driver);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get driver by login " + login, e);
+        }
+    }
+
+    private Driver getDriverWithLoginAndPassword(ResultSet resultSet) throws SQLException {
```

Comment:
you should return drivers with login and password from all methods
so let's refactor `getDriver()` method instead

we also should refactor `update()` method